### PR TITLE
Temp/user products

### PR DESCRIPTION
**Pull request by Yshah112**

### CODE REVIEW

File: products/urls.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
 from django.urls import path
 from django.conf.urls.static import static
 from django.conf import settings
-from .views import home_page
-
+from .views import home_page, category_data, \
```
Comment (snehals21) [inline products/urls.py]: @Yshah112 you can use tuple to multiple imports.

File: products/urls.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+    subcategory_data, product_data, all_products
 
 urlpatterns = [
     path('', home_page, name='homepage'),
     path('profile/', home_page, name='admin-profile'),
```
Comment (snehals21) [inline products/urls.py]: @Yshah112 give proper naming convensions to urls. here you have added `products` in 2 urls.
   ```
 path('product/<int:pk>/', product_data, name='products'),
 path('products/', all_products, name='all-products')
```
cc: @lalitvasoya

File: products/urls.py (hunk 1 above)
Comment (Yshah112) [reply products/urls.py]: done

File: products/urls.py (hunk 2 above)
Comment (Yshah112) [reply products/urls.py]: done

File: products/views.py [hunk 3]
```diff
@@ -14,6 +14,36 @@
...
+    """" To redirect user to home page and superuser to dashboard """
+
+    if not request.user.is_superuser:
+        category = Category.objects.filter(parent=None)
+        product = Product.objects.all()
+        if request.GET.get('search'):
+            category = category.filter(name__icontains=request.GET.get('search'))
+            product = product.filter(name__icontains=request.GET.get('search'))
+
+        return render(request, 'index.html', {'categorydata': category, 'productdata': product})
```
Comment (palakparmartntra) [inline products/views.py:31]: This will change after section management module. @rinkishivani @Yshah112

File: products/views.py [hunk 4]
```diff
@@ -358,3 +364,67 @@ def delete_brand(request, pk):
             'brand': brand
         }
         return render(request, 'product/brand/confirm_delete.html', context)
+
+
+def category_data(request):
```
Comment (palakparmartntra) [inline products/views.py:372]: Add docstring @Yshah112

File: products/views.py [hunk 5]
```diff
@@ -358,3 +364,67 @@ def delete_brand(request, pk):
...
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
+
+    return render(request, 'user_product/category.html', {'categorydata': page_obj})
+
+
+def subcategory_data(request, pk):
```
Comment (palakparmartntra) [inline products/views.py:391]: here as well  @Yshah112

File: products/views.py [hunk 6]
```diff
@@ -358,3 +364,67 @@ def delete_brand(request, pk):
...
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
+    return render(request, "user_product/subcategory.html", {'subcategorydata': page_obj})
+
+
+def product_data(request, pk):
+    id_parent = get_object_or_404(Category, pk=pk)
```
Comment (palakparmartntra) [inline products/views.py:414]: here as well  @Yshah112

File: products/views.py [hunk 7]
```diff
@@ -358,3 +364,67 @@ def delete_brand(request, pk):
...
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
+
+    return render(request, 'user_product/products.html', {'productdata': page_obj})
+
+
+def all_products(request):
```
Comment (palakparmartntra) [inline products/views.py:430]: here as well  @Yshah112

File: templates/user_product/products.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            </div>
+                  <div class="col-lg-12 ">
+        <div class="row ">
+                {% for product in productdata %}
+                <div class="col-lg-3">
+                    <div class="product-item">
+                        <div class="product-title">
+                            <a href="#">{{product.name}}</a>
+                            <div class="ratting">
+                                <i class="fa fa-star"></i>
```
Comment (palakparmartntra) [inline templates/user_product/products.html:19]: do we need this div class @Yshah112 ?

File: products/views.py (hunk 4 above)
Comment (Yshah112) [reply products/views.py:372]: done

File: products/views.py (hunk 5 above)
Comment (Yshah112) [reply products/views.py:391]: done

File: products/views.py (hunk 6 above)
Comment (Yshah112) [reply products/views.py:414]: done

File: products/views.py (hunk 7 above)
Comment (Yshah112) [reply products/views.py:430]: done

Comment (Yshah112) [reply products/views.py:430]: done

**Review by pratikbariya — APPROVED** [review]

**Pull request merged**